src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -385,7 +407,62 @@ Status GcsActorManager::RegisterActor(
     return Status::OK();
   }
 
-  auto actor = std::make_shared<GcsActor>(request);
+  // Mark the callback as pending and invoke it after the actor has been successfully
+  // created.
+  actor_to_register_callbacks_[actor_id].emplace_back(std::move(callback));
+
+  auto actor = std::make_shared<GcsActor>(request.task_spec());
+  RAY_CHECK(registered_actors_.emplace(actor->GetActorID(), actor).second);
+
+  const auto &owner_address = actor->GetOwnerAddress();
+  auto node_id = ClientID::FromBinary(owner_address.raylet_id());
+  auto worker_id = WorkerID::FromBinary(owner_address.worker_id());
+  RAY_CHECK(unresolved_actors_[node_id][worker_id].emplace(actor->GetActorID()).second);
+
+  return gcs_table_storage_->ActorTable().Put(
```

Comment:
Can you write a comment this Put will already succeed because the storage will be reliable?